signature/src/image/digest.rs

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::convert::TryFrom;
+use std::error::Error;
+use std::fmt;
+use std::str::FromStr;
+use crate::strum::EnumProperty;
+use std::string::ToString;
+
+// Supported digest algorithm types
+#[derive(EnumString, Display, Debug, PartialEq, EnumProperty)]
+pub enum Algorithm {
+    #[strum(serialize = "sha256", props(Length = "64"))]
+    Sha256,
+    #[strum(serialize = "sha384", props(Length = "96"))]
+    Sha384,
+    #[strum(serialize = "sha512", props(Length = "128"))]
+    Sha512,
+}
+
+// Reasons that parsing a string as a Digest can fail.
+#[derive(Debug, PartialEq, Eq)]
+pub enum ParseError {
+    /// Invalid checksum digest format
+    DigestInvalidFormat,
+    /// Invalid checksum digest length
+    DigestInvalidLength,
+    /// Unsupported digest algorithm
+    DigestUnsupported,
+}
+
+impl fmt::Display for ParseError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ParseError::DigestInvalidFormat => write!(f, "invalid checksum digest format"),
+            ParseError::DigestInvalidLength => write!(f, "invalid checksum digest length"),
+            ParseError::DigestUnsupported => write!(f, "unsupported digest algorithm"),
```

Comment:
Nit: my suggestion is to rename DigestUnsupported to DigestUnsupportedAlgorithm so we know exactly what's not supported in the digest.